src/main/java/io/mosip/mimoto/service/impl/IssuersServiceImpl.java

```diff
@@ -61,4 +119,138 @@ public IssuerDTO getIssuerConfig(String issuerId) throws ApiNotAccessibleExcepti
             issuerDTO = issuerConfigResp.get();
         return issuerDTO;
     }
+
+    @Override
+    public IssuerSupportedCredentialsResponse getCredentialsSupported(String issuerId, String search) throws ApiNotAccessibleException, IOException {
+        IssuerSupportedCredentialsResponse supportedCredentialsWithAuthorizationEndPoint = new IssuerSupportedCredentialsResponse();
+
+        IssuersDTO issuersDto = getAllIssuersWithAllFields();
+        Optional<IssuerDTO> issuerConfigResp = issuersDto.getIssuers().stream()
+                .filter(issuer -> issuer.getCredential_issuer().equals(issuerId))
+                .findFirst();
+        if (issuerConfigResp.isPresent()) {
+            IssuerDTO issuerDto = issuerConfigResp.get();
```

Comment:
No need to create issuerConfigResp object and use get for optional value. We can use optional chaining to filter and use filtered response using map.